src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -4,51 +4,77 @@
 
 namespace ray {
 
+// Utility function to return dependencies for the provided task spec that have
+// TaskTransportType::RAYLET.
+//
+std::vector<ObjectID> RayletDependencies(const TaskSpecification &task_spec) {
+  std::vector<ObjectID> dependencies;
+  for (size_t i = 0; i < task_spec.NumArgs(); i++) {
+    // NOTE: Multi-args are not currently supported.
+    if (task_spec.ArgIdCount(i) > 0) {
+      const ObjectID &dependency = task_spec.ArgId(i, 0);
+      if (dependency.GetTransportType() ==
+          static_cast<uint8_t>(TaskTransportType::RAYLET)) {
+        dependencies.push_back(dependency);
+      }
+    }
+  }
+  return dependencies;
+}
+
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   resolver_.ResolveDependencies(task_spec, [this, task_spec]() {
-    // TODO(ekl) should have a queue per distinct resource type required
     absl::MutexLock lock(&mu_);
-    queued_tasks_.push_back(task_spec);
-    RequestNewWorkerIfNeeded(task_spec);
+    // Only consider dependencies in plasma for scheduling tasks to workers
+    // (denoted by TaskTransportType::RAYLET). Note that the correct transport types
```

Comment:
We actually enforce the invariant that all these dependencies are RAYLET type after resolution. So the check is unnecessary (and probably conflicts with @stephanie-wang 's PR).

The only thing necessary is to re-compute the dependencies since the task may have been mutated.

Reply:
Nice, didn't catch that.